test/unit/test_dss_docker_image.py

```diff
@@ -96,3 +98,27 @@ def test_entrypoint_with_copy_args():
         "--notebooks", final,
         "--jupyter-server", jupyter,
     ]
+
+@pytest.fixture
+def mocked_docker_image():
+    testee = DssDockerImage("org/sample_repo", "version")
+    testee._start = datetime.now()
+    testee._start_container = MagicMock()
+    testee._install_dependencies = MagicMock()
+    testee._cleanup = MagicMock()
+    image = Mock(attrs={"Size": 1025})
+    testee._commit_container = MagicMock(return_value=image)
+    return testee
+
+
+def test_push_called(mocker, mocked_docker_image):
+    testee = mocked_docker_image
+
+    registry = DockerRegistry("user", "password")
+    registry.push = MagicMock()
```

Comment:
you can use registry = create_autospec(DockerRegistry), don't use workarounds everywhere only because you used it once.